crates/c-api/src/lib.rs

```diff
@@ -787,7 +787,10 @@ pub unsafe extern "C" fn wasm_module_new(
 ) -> *mut wasm_module_t {
     let binary = (*binary).as_slice();
     let store = &(*store).store.borrow();
-    let module = Module::new_unchecked(store, binary).expect("module");
+    let module = match Module::from_binary_unchecked(store, binary) {
```

Comment:
This is pre-existing, but while we're here, the resolution of https://github.com/WebAssembly/wasm-c-api/issues/123 was that `wasm_module_new` should indeed perform validation, so this should use `from_binary` rather than `from_binary_unchecked`.

Reply:
Heh I actually tried to switch to `from_binary`, but ran into https://github.com/bytecodealliance/wasmtime/issues/859 so we'll have a few other things to fix before switching over.